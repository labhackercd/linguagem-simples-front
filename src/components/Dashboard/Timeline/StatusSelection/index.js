import React, {useState} from 'react';
import {Paper,Button, TextField, Dialog, DialogActions, DialogContent, DialogContentText,
	      DialogTitle, Grid, Typography, Box,  List, ListItem} from '@material-ui/core';
import {makeStyles} from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
	summaryBox: {
		display: 'flex',
		flexDirection: 'row',
		width: '100%',
	},
	summaryHeader: {
  	display: 'flex',
		margin: '1rem 0 1rem 0',
  },
  tabs: {
    display: 'flex',
    minHeight: '0px',
    margin: '0.5rem 0 0 0',
    overflow: 'auto',
    flexWrap: 'no-wrap',
  },
  tab: {
    whiteSpace: 'nowrap',
    padding: '0 0.5rem',
    margin: '0 0.3rem',
    backgroundColor: '#F2F2F2',
    borderRadius: '5px',
    minHeight: '0px',
    minWidth: 'auto',
    textTransform: 'capitalize'
  },
}))
export default function StatusSelection(props) {
  const classes = useStyles();
  const [titlesArray] = useState([
                                  'Pré-sessão',
                                  'Sessão Iniciada',
                                  'Votação Iniciada',
                                  'Votação Encerrada',
                                  'Próxima Pauta',
                                  'Sessão Encerrada',
                                  'Fim da transmissão'])
  return (
    <Grid container className={classes.summaryBox}>
      <Grid container className={classes.summaryHeader} style={{margin: '1rem 0 0 0'}}>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-start'}}>
          <Typography variant="h5"> Status </Typography>
        </Grid>
        <Grid item md={6} style={{display:'flex', justifyContent: 'flex-end'}}>
        </Grid>
      </Grid>
      <Grid className={classes.tabs}>
          {titlesArray.map(function(title, index){
          return <Button key={index} uvariant="contained"
												 onClick={(e) => props.startUpdateWithTitleFlow(e, title)}
												 disableElevation
												 disabled={!props.broadcastingOnline}
												 className={classes.tab}>{title}</Button>
        })}
      </Grid>
    </Grid>
  )
}
