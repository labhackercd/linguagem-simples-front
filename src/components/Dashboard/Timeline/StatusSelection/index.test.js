import React from 'react';
import { render, screen } from '@testing-library/react';
import StatusSelection from './index';
import {shallow} from "enzyme/build";

it("should render the status selection section", () => {
    const component = shallow(<StatusSelection/>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});
