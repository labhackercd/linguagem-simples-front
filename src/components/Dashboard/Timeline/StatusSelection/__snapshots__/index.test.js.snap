// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should not have changes 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-summaryBox-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-summaryHeader-2"
    container={true}
    style={
      Object {
        "margin": "1rem 0 0 0",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-start",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
         Status 
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={6}
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-tabs-3"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="0"
      onClick={[Function]}
      uvariant="contained"
    >
      Pré-sessão
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="1"
      onClick={[Function]}
      uvariant="contained"
    >
      Sessão Iniciada
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="2"
      onClick={[Function]}
      uvariant="contained"
    >
      Votação Iniciada
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="3"
      onClick={[Function]}
      uvariant="contained"
    >
      Votação Encerrada
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="4"
      onClick={[Function]}
      uvariant="contained"
    >
      Próxima Pauta
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="5"
      onClick={[Function]}
      uvariant="contained"
    >
      Sessão Encerrada
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      className="makeStyles-tab-4"
      disableElevation={true}
      disabled={true}
      key="6"
      onClick={[Function]}
      uvariant="contained"
    >
      Fim da transmissão
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
